src/features/filter.ts

```diff
@@ -1,10 +1,54 @@
-export const actions = { /* put action creators here */};
+import { Status } from '../types/Status';
 
-const filterReducer = () => {
-  return {
-    query: '',
-    status: 'all',
-  };
+type SetQueryAction = {
+  type: 'query/SET';
+  payload: string;
+};
+
+type SetStatusAction = {
+  type: 'status/SET';
+  payload: Status;
+};
+
+const setQuery = (payload: string): SetQueryAction => ({
+  type: 'query/SET',
+  payload,
+});
+
+const setStatus = (payload: Status): SetStatusAction => ({
+  type: 'status/SET',
+  payload,
+});
+
+export const actions = { setQuery, setStatus };
+
+type State = {
+  query: string;
+  status: Status;
+} | null;
+
+const InitialState = { query: '', status: 'all' } as State;
```

Comment:
```suggestion
const initialState = { query: '', status: 'all' } as State;
```
Don't name regular variables with the first capital letter.